### ISSUE TITLE
Non free firmware-ralink conflicts with armbian-firmware

### ISSUE THREAD

**Issue by zamentur** [labriqueinternet/hotspot_ynh#30]
I had tried to setup hotspot on an armbian (The version for orangepi pc + that ARN uses for create the NeutriBox).
There are errors:
```
# apt full-upgrade
Reading package lists... Done
Building dependency tree       
Reading state information... Done
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 firmware-linux-nonfree : Depends: firmware-misc-nonfree (= 20161130-4) but it is not installed
 firmware-ralink : Depends: firmware-misc-nonfree but it is not installed
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).

root@arn20:~# apt --fix-broken install
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following additional packages will be installed:
  firmware-misc-nonfree
The following NEW packages will be installed:
  firmware-misc-nonfree
0 upgraded, 1 newly installed, 0 to remove and 5 not upgraded.
28 not fully installed or removed.
Need to get 0 B/2 320 kB of archives.
After this operation, 8 155 kB of additional disk space will be used.
Do you want to continue? [O/n] 
(Reading database ... 44787 files and directories currently installed.)
Preparing to unpack .../firmware-misc-nonfree_20161130-4_all.deb ...
Unpacking firmware-misc-nonfree (20161130-4) ...
dpkg: error processing archive /var/cache/apt/archives/firmware-misc-nonfree_20161130-4_all.deb (--unpack):
 trying to overwrite '/lib/firmware/mt7601u.bin', which is also in package armbian-firmware 5.65
dpkg-deb: error: subprocess paste was killed by signal (Broken pipe)
Errors were encountered while processing:
 /var/cache/apt/archives/firmware-misc-nonfree_20161130-4_all.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)

root@arn20:~# apt list --upgradable
Listing... Done
linux-dtb-next-sunxi/stretch 5.70 armhf [upgradable from: 5.67]
linux-image-next-sunxi/stretch 5.70 armhf [upgradable from: 5.67]
linux-libc-dev/stretch 5.70 armhf [upgradable from: 4.9.130-2]
linux-stretch-root-next-orangepipcplus/stretch 5.70 armhf [upgradable from: 5.65]
linux-u-boot-orangepipcplus-next/stretch 5.70 armhf [upgradable from: 5.65]


```

**Comment by zamentur** [comment]
Note the hotspot works but after that i am not able to upgrade the NeutriBox... The solution is to not install nonfree firmware (but in my case it's the script on the internetcube that have configure that)
https://github.com/labriqueinternet/hotspot_ynh/blob/master/scripts/_common.sh#L8

To fix that i have done:
```
dpkg -r firmware-ralink
apt remove firmware-linux-nonfree
apt update
apt full-upgrade
```

**Comment by keomabrun** [comment]
Running the installation process with `firmware_nonfree=no` in the hypercube file seem to work.
The Wifi antenna is well detected however the access point does not seem to be created.

Iwconfig output:
```
# iwconfig wlx6cfdb9083a3b 
wlx6cfdb9083a3b  IEEE 802.11  ESSID:off/any  
          Mode:Managed  Access Point: Not-Associated   Tx-Power=20 dBm   
          Retry short limit:7   RTS thr:off   Fragment thr:off
          Encryption key:off
          Power Management:off
```

Hostpot App Output (from Web UI)
```
    [INFO] Autodetected internet interface: tun0 (last start: tun0)
    [OK] IPv4 NAT set
    [OK] IPv6/IPv4 forwarding set
    [OK] Hostapd is running
    [INFO] hotspot0: No IPv6 delegated prefix found
    [OK] hotspot0: DHCPv4 server (dnsmasq) is running
    [OK] hotspot0: IPv4 NAT address set
```

Installation log file: https://chif.fr/nextcloud/index.php/s/QazxoeSjcr4FZan

**Comment by alexAubin** [comment]
Fixed by #54
